### ISSUE TITLE
Index representatives and commenters on AOs

### ISSUE THREAD

**Issue by noahmanger** [18F/openFEC#2240]
So that users can eventually search AOs by the names of the counsels or commenters on a given AO, index the necessary fields on each AO case. 

---

We currently have the ability to search AOs by "requestor". The purpose of this story is to index and expose the "commenter" and "representative" (or "counsel"?) fields. 

**Completion criteria:**
- [ ] The AO endpoint can be filtered by `requestor`, accepting multiple parameters and returning results that match X `OR` Y
- [ ] The AO endpoint can be filtered by `representative`, accepting multiple parameters and returning results that match X `OR` Y

**Comment by dogweather** [comment]
I'm not very familiar with this codebase; I pinging @ccostino for help where to begin.

**Comment by ccostino** [comment]
Unfortunately I'm not going to be much help here, as I haven't worked in the legal docs code myself at any great length.  @anthonygarvan and @vrajmohan are going to be far more help!  From what I can tell, though, the places to start looking would be here:

* [`webservices/legal_docs/advisory_opinions.py`](https://github.com/18F/openFEC/blob/develop/webservices/legal_docs/advisory_opinions.py): code specific to the AOs (Advisory Opinions)
* [`webservices/legal_docs/load_legal_docs.py`](https://github.com/18F/openFEC/blob/develop/webservices/legal_docs/load_legal_docs.py): code related to loading the legal docs into the Elasticsearch indexes
* [`webservices/resources/legal.py`](https://github.com/18F/openFEC/blob/develop/webservices/resources/legal.py): code related to the legal docs endpoints.

Does this sound right to you, @anthonygarvan or @vrajmohan?  Would one of you be able to pair with @dogweather to help with getting a start since this is uncharted territory for me as well?  Thanks!

**Comment by vrajmohan** [comment]
I can help.

**Comment by dogweather** [comment]
Thanks! I'm starting with https://github.com/18F/openFEC/wiki/Create-an-endpoint as a roadmap.

**Comment by vrajmohan** [comment]
When we index requestors, we also index (and allow) search on the requestor type, which is one of:
```
{1: 'Federal candidate/candidate committee/officeholder',
 2: 'Publicly funded candidates/committees',
 3: 'Party committee, national',
 4: 'Party committee, state or local',
 5: 'Nonconnected political committee',
 6: 'Separate segregated fund',
 7: 'Labor Organization',
 8: 'Trade Association',
 9: 'Membership Organization, Cooperative, Corporation W/O Capital Stock',
 10: 'Corporation (including LLCs electing corporate status)',
 11: 'Partnership (including LLCs electing partnership status)',
 12: 'Governmental entity',
 13: 'Research/Public Interest/Educational Institution',
 14: 'Law Firm',
 15: 'Individual',
 16: 'Other'}
```
Are we interested in the types of commenters and representatives?

**Comment by noahmanger** [comment]
Paging @nickykrause . I know we addressed this question before, and I can dig up the answer, but maybe you know off the top of your head?

**Comment by nickykrause** [comment]
I don't recall users having a notable interest in commenter types and representative types. Also, in SAOS, it does appear to be possible to search by commenter/representative type (although it is possible to search by requestor type).

So, I would say that we don't need it

**Comment by vrajmohan** [comment]
Fixed by 18F/openFEC#2316
